Created by Franky Backeljauw (Universiteit Antwerpen)
Install header files and scripts
Rename static lib to freebayes, and set install to True
Add target to build shared library
--- freebayes-1.3.10.orig/meson.build	2026-09-04 13:51:50.401569239 +0200
+++ freebayes-1.3.10/meson.build	2026-09-04 14:44:27.171876129 +0200
@@ -156,6 +156,37 @@
 freebayes_common_src += vcflib_src
 freebayes_common_src += smithwaterman_src
 
+# FreeBayes headers
+freebayes_headers = files(
+    'src/Allele.h',
+    'src/AlleleParser.h',
+    'src/BedReader.h',
+    'src/Bias.h',
+    'src/CNV.h',
+    'src/Contamination.h',
+    'src/DataLikelihood.h',
+    'src/Dirichlet.h',
+    'src/Ewens.h',
+    'src/FBFasta.h',
+    'src/Genotype.h',
+    'src/IndelAllele.h',
+    'src/LeftAlign.h',
+    'src/Logging.h',
+    'src/Marginals.h',
+    'src/Multinomial.h',
+    'src/NonCall.h',
+    'src/Parameters.h',
+    'src/Product.h',
+    'src/Result.h',
+    'src/ResultData.h',
+    'src/Sample.h',
+    'src/SegfaultHandler.h',
+    'src/Sum.h',
+    'src/TryCatch.h',
+    'src/Utility.h',
+    'src/version_git.h',
+    )
+
 # Include paths
 incdir = include_directories(
     'src',
@@ -171,16 +202,30 @@
     )
 
 freebayes_lib = static_library(
-    'freebayes_common',
+    'freebayes',
     [freebayes_common_src, fastahack_src],
     pic: false,
     include_directories : [incdir, vcflib_inc, smithwaterman_inc, fastahack_inc],
     cpp_args : extra_cpp_args,
     dependencies : [vcflib_dep, zlib_dep, lzma_dep, thread_dep, htslib_dep, tabixpp_dep,
                     wfa2lib_dep, seqlib_dep],
-    install : false,
+    install : true,
     )
 
+freebayes_slib = shared_library(
+    'freebayes',
+    freebayes_common_src,
+    include_directories : incdir,
+    cpp_args : extra_cpp_args,
+    dependencies : [zlib_dep, lzma_dep, thread_dep, htslib_dep, tabixpp_dep,
+                    vcflib_dep, seqlib_dep, fastahack_dep, smithwaterman_dep],
+    install : true,
+    )
+
+install_headers(freebayes_headers, subdir : 'freebayes')
+
+install_subdir('scripts', install_dir : '')
+
 if static_build
   link_arguments = '-static'
 else
@@ -203,7 +248,7 @@
                           fastahack_dep, # for vcflib
                           smithwaterman_dep # for vcflib
                           ],
-           link_with : [freebayes_lib],
+           link_with : [freebayes_slib],
            install: true
           )
 
@@ -221,7 +266,7 @@
                           fastahack_dep, # for vcflib
                           smithwaterman_dep, # for vcflib
                           wfa2lib_dep],
-           link_with : freebayes_lib,
+           link_with : freebayes_slib,
            install: true
           )
